### PR TITLE
Maintenance updates

### PR DESCRIPTION
**Pull request by tylerkelly13**
Changes:
- Switched to yarn
- Updated packages to resolve security issues, removed unused packages, move to maintained alternative packages where possible, locked package versions
- converted double quotes to single quotes where possible
- converted hard-coded tabs to spaces

### CODE REVIEW

File: lib/moxiedoc.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var Parser = require('./parser').Parser;
 var Builder = require('./builder').Builder;
 var Exporter = require('./exporter').Exporter;
 var reporter = require('./reporter');
-var wildcard = require('wildcard');
+var matcher = require('matcher');
```
Comment (lnewson) [inline lib/moxiedoc.js:5]: I assume there was an issue with the `wildcard` library?

File: package.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
   "description": "Documentation tool",
   "author": "Johan Sörlin <spocke@moxiecode.com>",
   "bugs": {
-    "url": "http://github.com/spocke/moxiedoc/issues"
+    "url": "https://github.com/tinymce/moxiedoc/issues"
   },
   "contributors": [],
   "engines": {
-    "node": ">=0.8.0"
+    "node": ">=14.0.0"
```
Comment (lnewson) [inline package.json:11]: This is a breaking change and could cause issues. I'd suggest being a little more lenient here and making this `>=12.0.0` (the previous LTS version). Also given it's a breaking change, it should have a major version bump, however this is a `0.x` release so maybe we should just do a minor instead? eg 0.1.0

Comment (lnewson) [reply package.json:11]: I should also add I've not seen anything that actually requires a new version of NodeJS, so it doesn't really make sense to be changing this.

File: templates/tinymcenext/template.js [hunk 3]
```diff
@@ -24,102 +23,102 @@ var namespaceDescriptions = {
...
+  // sort types alphabetically
+  var sortedTypes = root.getTypes().sort(function (a, b) {
+    if (a.fullName < b.fullName) return -1;
+    if (a.fullName > b.fullName) return 1;
+    return 0;
+  });
+
+  addPageToArchive({
+    filename: '_data/nav_api.yml',
+    content: YAML.dump(getNavFile(sortedTypes))
```
Comment (lnewson) [inline templates/tinymcenext/template.js:43]: I assume this was needed due to the js-yaml 3 -> 4 conversion?

**Review by lnewson — APPROVED** [review]

File: lib/moxiedoc.js (hunk 1 above)
Comment (tylerkelly13) [reply lib/moxiedoc.js:5]: https://www.npmjs.com/package/wildcard
> NOTE
> Work on this project is largely inactive, now so I'd recommend checking out the wonderful matcher package as a solid alternative.

no problem, just moving to an actively maintained alternative

File: templates/tinymcenext/template.js (hunk 3 above)
Comment (tylerkelly13) [reply templates/tinymcenext/template.js:43]: https://github.com/nodeca/js-yaml/blob/master/migrate_v3_to_v4.md#safeload-safeloadall-safedump--load-loadall-dump
> Older safe* functions were renamed.

They dropped the "safe" prefixes

File: package.json (hunk 2 above)
Comment (tylerkelly13) [reply package.json]: Minor version bump and reduced nodejs version in https://github.com/tinymce/moxiedoc/pull/3/commits/0cb3456fcffb83f14aaca1998038cefacc650336

File: package.json [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
   "description": "Documentation tool",
   "author": "Johan Sörlin <spocke@moxiecode.com>",
   "bugs": {
-    "url": "http://github.com/spocke/moxiedoc/issues"
+    "url": "https://github.com/tinymce/moxiedoc/issues"
   },
   "contributors": [],
   "engines": {
-    "node": ">=0.8.0"
+    "node": ">=12.0.0"
```
Comment (tylerkelly13) [inline package.json:11]:
```suggestion
    "node": ">=0.8.0"
```

File: package.json (hunk 2 above)
Comment (tylerkelly13) [reply package.json]: Reversed this to 0.8.0 to avoid updating docker image at this time.

**Pull request merged**